tendermint/src/lite/verifier.rs

```diff
@@ -330,349 +333,296 @@ where
+        commit: Vec<usize>,
+        err: Error,
+    ) {
         let (un_sh, un_vals, un_next_vals) = next_state(vals, commit);
-        let result = verify_single(ts, &un_sh, &un_vals, &un_next_vals, THRESHOLD);
+        let result = verify_single_inner(
+            ts,
+            &un_sh,
+            &un_vals,
+            &un_next_vals,
+            TrustThresholdFraction::default(),
+        );
         assert_eq!(result, Err(err));
     }
 
     // make a state with the given vals and commit and ensure we get no error.
-    fn assert_ok(ts: &MockState, vals: Vec<usize>, commit: Vec<Option<usize>>) {
+    fn assert_single_ok(ts: &MockState, vals: Vec<usize>, commit: Vec<usize>) {
         let (un_sh, un_vals, un_next_vals) = next_state(vals, commit);
-        assert!(verify_single(ts, &un_sh, &un_vals, &un_next_vals, THRESHOLD).is_ok());
+        assert!(verify_single_inner(
+            ts,
+            &un_sh,
+            &un_vals,
+            &un_next_vals,
+            TrustThresholdFraction::default()
+        )
+        .is_ok());
+    }
+
+    // make a sequence of states with the given vals for the requester
+    // and ensure bisection yields no error.
+    fn assert_bisection_ok(
+        req: &MockRequester,
+        ts: &TrustedState<MockCommit, MockHeader>,
+        untrusted_height: u64,
+        expected_num_of_requests: usize,
+    ) {
+        let mut cache: Vec<MockTrustedState> = Vec::new();
+        let ts_new = verify_bisection_inner(
+            &ts,
+            untrusted_height,
+            TrustThresholdFraction::default(),
+            req,
+            cache.as_mut(),
+        )
+        .expect("should have passed");
+        assert_eq!(ts_new.last_header().header().height(), untrusted_height);
+        assert_eq!(cache.len(), expected_num_of_requests);
+        assert_uniqueness(cache);
+    }
+
+    fn assert_uniqueness(cache: Vec<TrustedState<MockCommit, MockHeader>>) {
+        let mut uniq = cache.clone();
+        uniq.dedup();
+        assert_eq!(cache, uniq);
     }
 
     // convenience vars for validators that signed commit
-    static S0: Option<usize> = Some(0);
-    static S1: Option<usize> = Some(1);
-    static S2: Option<usize> = Some(2);
-    static S3: Option<usize> = Some(3);
-    static S4: Option<usize> = Some(4);
-    static S5: Option<usize> = Some(5);
+    static S0: usize = 0;
+    static S1: usize = 1;
+    static S2: usize = 2;
+    static S3: usize = 3;
+    static S4: usize = 4;
+    static S5: usize = 5;
 
     // valid to skip, but invalid commit. 1 validator.
     #[test]
     fn test_verify_single_skip_1_val_verify() {
-        let ts = &init_state(vec![0]);
+        let ts = &init_trusted_state(vec![0], vec![0], 1);
 
         // 100% overlap, but wrong commit.
         // NOTE: This should be an invalid commit error since there's
-        // a vote from a validator not in the set !
-        // but voting_power_in isn't smart enough to see this ...
```

Comment:
So this comment is still relevant in another case, for instance `assert_err(ts, vec![1], vec![S0, S1], Error::InvalidCommit)`, but here it's an InvalidCommit because the commit includes a signature from someone not in the validator set (S0). But voting_power_in still can't detect that!

Reply:
Added back the comment for now which links the tracking issue.